src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ObjectModelConverters.cs

```diff
@@ -238,22 +215,88 @@ internal static void FixUpTestCase(this TestCase testCase, string? testAssemblyP
         testCase.ExecutorUri = new(Constants.ExecutorUri);
     }
 
-    private static bool TryParseFullyQualifiedType(string fullyQualifiedName, [NotNullWhen(true)] out string? fullyQualifiedType)
+    private static bool TryGetMethodIdentifierProperty(TestCase testCase, [NotNullWhen(true)] out TestMethodIdentifierProperty? methodIdentifierProperty)
     {
-        fullyQualifiedType = null;
+        string? managedType = testCase.GetPropertyValue<string>(ManagedTypeProperty, defaultValue: null);
+        string? managedMethod = testCase.GetPropertyValue<string>(ManagedMethodProperty, defaultValue: null);
+        // NOTE: ManagedMethod, in case of MSTest, will have the parameter types.
+        // So, we prefer using it to display the parameter types in Test Explorer.
+        return !RoslynString.IsNullOrEmpty(managedType) && !RoslynString.IsNullOrEmpty(managedMethod)
+            ? TryGetMethodIdentifierPropertyFromManagedTypeAndManagedMethod(managedType, managedMethod, out methodIdentifierProperty)
+            : TryGetMethodIdentifierPropertyFromFullyQualifiedName(testCase.FullyQualifiedName, out methodIdentifierProperty);
+    }
+
+    private static bool TryGetMethodIdentifierPropertyFromManagedTypeAndManagedMethod(
+        ReadOnlySpan<char> managedType,
+        ReadOnlySpan<char> managedMethod,
+        [NotNullWhen(true)] out TestMethodIdentifierProperty? methodIdentifierProperty)
+    {
+        string assemblyFullName = string.Empty;
+        string @namespace;
+        string typeName;
+        string methodName;
+        string[] parameterTypeFullNames = Array.Empty<string>();
```

Comment:
avoid redundantly assigning the variable. and instead assign it in the else of the below of

Reply:
perhaps use a collection expression instead of `Array.Empty<string>();`